### PR TITLE
 fix comment parsing at start of file

### PR DESCRIPTION
**Pull request by ajafff**
* skip shebang if present (fixes: #28477)
* ~~don't parse trailing comments to avoid duplicates that are also returned as leading comments~~

### CODE REVIEW

File: src/compiler/scanner.ts [hunk 1]
```diff
@@ -642,8 +642,18 @@ namespace ts {
         let pendingKind!: CommentKind;
         let pendingHasTrailingNewLine!: boolean;
         let hasPendingCommentRange = false;
-        let collecting = trailing || pos === 0;
+        let collecting = trailing;
         let accumulator = initial;
+        if (pos === 0) {
+            if (collecting) {
```
[the remark this answers is not in the document]
Comment (ajafff) [reply src/compiler/scanner.ts]: While writing the tests I noticed that comments in the first line of a file were returned as leading **and** trailing comment. So I made this small change to never return trailing comments at position 0

[the remark this answers is not in the document]
Comment (ajafff) [reply src/compiler/scanner.ts]: @sheetalkamat would you prefer these changes to be split into two PRs (one to skip the shebang and one to not parse trailing comments at the start of the file)?

[the remark this answers is not in the document]
Comment (rbuckton) [reply src/compiler/scanner.ts]: @ajafff I don't think we need the additional change for now, unless there's a test case that shows this being an issue.

[the remark this answers is not in the document]
Comment (rbuckton) [reply src/compiler/scanner.ts]: I believe the reason this hasn't been necessary is that `forEachLeadingCommentToEmit` and `forEachTrailingCommentToEmit` in emitter.ts handle not emitting overlapping comment ranges. I don't think this change is necessary for now and we should stick to just skipping the shebang in this PR.

[the remark this answers is not in the document]
Comment (ajafff) [reply src/compiler/scanner.ts]: Ok, so I'll revert this change and open an issue to discuss changing it in a follow-up PR.

**Comment by rbuckton** [comment]
@RyanCavanaugh do we want to take this for 3.3? If not I'll merge this in the afternoon.

**Comment by rbuckton** [comment]
Thanks for the contribution!

**Pull request merged**